prov/efa/src/efa_user_info.c

```diff
@@ -302,28 +304,42 @@ bool efa_user_info_should_support_hmem(int version, const struct fi_info *hints)
 	 */
 	if (FI_VERSION_GE(version, FI_VERSION(1, 18)) && hmem_ops[FI_HMEM_CUDA].initialized) {
 		EFA_INFO(FI_LOG_CORE,
-			"User is using API version >= 1.18. CUDA library and device are availble, claim support of FI_HMEM.");
+			"User is using API version >= 1.18. CUDA library and "
+			"device are availble, claim support of FI_HMEM.");
 		return true;
 	}
 
 	if (ofi_hmem_p2p_disabled()) {
 		EFA_WARN(FI_LOG_CORE,
 			"FI_HMEM capability currently requires peer to peer "
-			"support, which is disabled because FI_HMEM_P2P_DISABLED was set to 1/on/true.");
+			"support, which is disabled because "
+			"FI_HMEM_P2P_DISABLED was set to 1/on/true.");
 		return false;
 	}
 
-	if (!efa_device_support_rdma_read()) {
+	if (!FI_VERSION_GE(version, FI_VERSION(1, 18))) {
+		extra_version_info = " the application is using libfabric "
```

Comment:
it seems we just need `extra_info`, and we can make it point to the two different message

Reply:
Done.  I also re-worked the return logic a little.  Please take a close look here.